packages/driver-azure/src/fs/index.ts

```diff
@@ -0,0 +1,89 @@
+import { DefaultAzureCredential } from '@azure/identity'
+import { VirtualFS } from '@preevy/core'
+
+export const defaultBucketName = (
+  { profileAlias, accountId }: { profileAlias: string; accountId: string },
+) => `preevy-${accountId}-${profileAlias}`
+
+const ensureBucketExists = async (blobServiceClient: BlobServiceClient, containerName: string) => {
+  const containerClient: ContainerClient = blobServiceClient.getContainerClient(containerName)
+  const exists = await containerClient.exists()
+
+  if (!exists) {
+    await containerClient.create()
+  }
+  return containerName
+}
+
+export const parseAzureBlobUrl = (azureBlobUrl: string) => {
+  const url = new URL(azureBlobUrl)
+
+  if (!(url.protocol === 'https:')) {
+    throw new Error('Azure Blob urls must start with https:')
+  }
+
+  const accountName = url.hostname.split('.')[0]
+
+  const pathSegments = url.pathname.split('/')
+  const containerName = pathSegments[1]
+
+  return {
+    url,
+    accountName,
+    containerName,
+  }
+}
+
+async function streamToBuffer(readableStream: NodeJS.ReadableStream) {
+  return await new Promise<Buffer>((resolve, reject) => {
+    const chunks: Buffer[] = []
+    readableStream.on('data', data => {
+      chunks.push(data instanceof Buffer ? data : Buffer.from(data))
+    })
+    readableStream.on('end', () => {
+      resolve(Buffer.concat(chunks))
+    })
+    readableStream.on('error', reject)
+  })
+}
+
+export const azureStorageBlobFs = async (azureBlobUrl: string): Promise<VirtualFS> => {
+  const { accountName, containerName } = parseAzureBlobUrl(azureBlobUrl)
+
+  const defaultAzureCredential = new DefaultAzureCredential()
+
+  const blobServiceClient = new BlobServiceClient(
+    `https://${accountName}.blob.core.windows.net`,
+    defaultAzureCredential
+  )
+  const containerClient = blobServiceClient.getContainerClient(containerName)
+
+  await ensureBucketExists(blobServiceClient, containerName)
+
+  return {
+    read: async (filename: string) => {
+      const blobClient = containerClient.getBlobClient(filename)
+      const result = await blobClient.download()
+      if (result.readableStreamBody !== undefined) {
+        return await streamToBuffer(result.readableStreamBody)
+      }
+      return undefined
+    },
+    write: async (filename: string, content: Buffer | string) => {
+      const blockBlobClient = containerClient.getBlockBlobClient(filename)
+      await blockBlobClient.upload(content, content.length)
+    },
+    delete: async (filename: string) => {
+      const blockBlobClient = containerClient.getBlockBlobClient(filename)
+      const options: BlobDeleteOptions = {
+        deleteSnapshots: 'include',
+      }
+      const blobDeleteIfExistsResponse: BlobDeleteIfExistsResponse = await blockBlobClient.deleteIfExists(options)
+
+      if (blobDeleteIfExistsResponse.errorCode) {
+        throw new Error(`Error: ${blobDeleteIfExistsResponse.errorCode}`)
```

Comment:
according to [this](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/storage-blob/test/blobclient.spec.ts#L344), the `errorCode` will be set if the blob is not found - the contract requires to not throw.